coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelNodeToCoralSqlNodeTransformer.java

```diff
@@ -0,0 +1,172 @@
+    correlTableMap.put(e.getCorrelationId(), leftResult.qualifiedContext());
+
+    final Result rightResult = visitChild(1, e.getRight());
+
+    SqlJoin join = new SqlJoin(POS, leftResult.asFrom(), SqlLiteral.createBoolean(false, POS),
+        JoinType.COMMA.symbol(POS), rightResult.asFrom(), JoinConditionType.NONE.symbol(POS), null);
+
+    SqlIdentifier leftNode = (SqlIdentifier) leftResult.node;
+    SqlBasicCall lateralRightNode = (SqlBasicCall) rightResult.node;
+    SqlBasicCall asSqlNode = (SqlBasicCall) lateralRightNode.getOperands()[0];
+    SqlIdentifier asSqlNodeAlias = (SqlIdentifier) asSqlNode.getOperands()[1];
+
+    //The default implementation assumes that the rightLateralNode, for LateralSqlOperator,
+    // has a null alias. At the same time, it expects a non-null alias for the same node.
+    // At this point, we can either customize the rightLateralNode to pass the alias check
+    // or we can bypass the check and work around it. The latter appoach has been chosen.
+    final ImmutableMap.Builder<String, RelDataType> builder =
+        ImmutableMap.<String, RelDataType> builder().put(Util.last(leftNode.names), leftResult.neededType)
+            .put(Util.last(asSqlNodeAlias.names), rightResult.neededType);
+
+    return new Result(join, Expressions.list(Clause.FROM), null, null, builder.build());
+  }
+
+  /**
+   * CORAL represents explode function as an Uncollect RelNode.
+   * The default super's implementation takes Project RelNode as input in the form of
+   *        SELECT `complex`.`c` as`ccol` FROM (VALUES  (0))
+   * and appends the UNNEST and AS operators on top of it.
+   *
+   * However, the result generated is unmanageable for the following reasons:
+   * 1. Result has some extra clauses with make parsing difficult.
+   * 2. It also doesn't work well for unboxing array of type struct.
+   * It attempts to generate individual columns for each datatype inside the struct.
+   * 3. It doesn't append the LATERAL operator
+   *
+   * Overriding the function helps generate a more easily parsable SqlNode:
+   * LATERAL UNNEST(`complex`.`c`) AS t0 (ccol)
+   */
+  @Override
+  public Result visit(Uncollect e) {
+
+    final Result x = visitChild(0, e.getInput());
+
+    // Extract unnestColumns from result x: (SELECT <unnestColumns> FROM (VALUES(0)))
+    // and generate UNNEST(<unnestColumns>) instead.
+    final List<SqlNode> unnestOperands = new ArrayList<>();
+    for (RexNode unnestCol : ((Project) e.getInput()).getChildExps()) {
+      unnestOperands.add(x.qualifiedContext().toSql(null, unnestCol));
+    }
+
+    Function unnestOperator = functionResolver.tryResolve("explode", null,
+        // The first element of sqlOperands is the operator itself. The actual # of operands is sqlOperands.size() - 1
+        unnestOperands.size() - 1);
+
+    final SqlNode unnestNode = unnestOperator.getSqlOperator().createCall(POS, unnestOperands.toArray(new SqlNode[0]));
```

Comment:
I would suggest to use 
```
final SqlNode unnestNode = (e.withOrdinality ? HivePosExplodeOperator.POS_EXPLODE : HiveExplodeOperator.EXPLODE)
        .createCall(POS, unnestOperands.toArray(new SqlNode[0]));
```
directly because
1. it would help to avoid using `HiveFunctionResolver` and `StaticHiveFunctionRegistry`
2. I think we also need to consider `posexplode` function, also need to add unit tests for that function

Reply:
1. Is there any particular reason we want to avoid using `HiveFunctionResolver` and `StaticHiveFunctionRegistry`. 
We use these components when moving from source language -> CoralRelNode. I was thinking we could re-leverage them when moving from CoralRelNode -> CoralSqlNode -> destination language.  What do you think?

2. This PR doesn't support posexplode yet. I plan to raise a separate (next) PR for it. If it's confusing, I can call that out in the PR description. What do you think?

Reply:
For 1, given this visitor method uses the specific `unnest` function, I think we don't need `HiveFunctionResolver` and `StaticHiveFunctionRegistry` in this visitor method. We may add them for general UDFs in other visitor methods in the future if needed.
For 2, I am fine with handling `posexplode` in a following PR, please also modify the PR summary to be `lateral view explode` given `posexplode` is also a kind of `lateral view`

Reply:
sounds good. I accommodated your recommendation for (1).
For (2), the latest draft now supports `posexplode` as well. I added a UT validating it as well. 
Resolving this issue.